Index: app/src/main/java/com/example/foodtok/services/IInteractionService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/foodtok/services/IInteractionService.java b/app/src/main/java/com/example/foodtok/services/IInteractionService.java
new file mode 100644
--- /dev/null	(date 1775493137181)
+++ b/app/src/main/java/com/example/foodtok/services/IInteractionService.java	(date 1775493137181)
@@ -0,0 +1,8 @@
+package com.example.foodtok.services;
+
+public interface IInteractionService {
+    void likeRecipe(String recipeId, InteractionCallback callback);
+    void saveRecipe(String recipeId, InteractionCallback callback);
+    void addComment(String recipeId, String text, InteractionCallback callback);
+
+}
